app/src/main/java/com/sdp13epfl2021/projmag/database/interfaces/UserdataDatabase.kt

```diff
@@ -152,4 +153,26 @@ interface UserdataDatabase {
         onSuccess: () -> Unit,
         onFailure: (Exception) -> Unit
     )
+
+    /**
+     * Uploads the user profile to a remote database
+     * @param profile the user profile to upload
+     * @param onSuccess the closure that's when a profile is uploaded successfully
+     * @param onFailure the closure that's called if the upload fails
+     */
+    fun uploadProfile(
+        profile: ImmutableProfile,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    )
+
+    /**
+     * Gets the user profile from the remote database if it exists
```

Comment:
It's not correct, because some implementation use a local cache/offline database.

Reply:
fixed